pkg/types/ccip/models.go

```diff
@@ -18,3 +63,40 @@ type TxMeta struct {
 	TxHash                  string
 	LogIndex                uint64
 }
+
+// EIP55 converts the provided address to EIP55.
+//
+// If the ith digit is a letter (ie. it’s one of abcdef) print it in uppercase if the 4*ith bit of the hash of the
+// lowercase hexadecimal address is 1 otherwise print it in lowercase.
+func EIP55(addr string) (string, error) {
```

Comment:
This function does multiple things right now - its basically our check for whether an address is an Ethereum address or not, and secondarily, it tries to convert it to EIP55 format.

Not necessarily bad but I'm wondering if we need more robust checks here to avoid e.g EIP-55'ing a Solana or Starknet address.

Reply:
cleaned this up